storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4553,6 +4561,19 @@ int ha_rocksdb::convert_record_to_storage_format(
   *packed_rec =
       rocksdb::Slice(m_storage_record.ptr(), m_storage_record.length());
 
+  // We've obtained the TTL timestamp for this record. Now store it in
+  // m_ttl_bytes if it wasn't already so we can update secondary indexes with
+  // this value as well.
+  if (has_ttl) {
+    if (!row_info.old_pk_slice.empty() && has_ttl_column) {
+      m_ttl_bytes_updated = memcmp(m_ttl_bytes, m_storage_record.ptr(),
+                                   ROCKSDB_SIZEOF_TTL_RECORD);
+    } else {
+      m_ttl_bytes_updated = false;
+    }
+    memcpy(m_ttl_bytes, m_storage_record.ptr(), ROCKSDB_SIZEOF_TTL_RECORD);
```

Comment:
Good point! The correct check is a bit more convoluted so I added a comment to clarify things.